remote-source/main.go

```diff
@@ -0,0 +1,277 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/auto"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto/optdestroy"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto/optrefresh"
+	"github.com/pulumi/pulumi/sdk/v3/go/auto/optup"
+	"gopkg.in/yaml.v3"
+)
+
+type Project struct {
+	Location string `yaml:"location"`
+	Name     string `yaml:"name"`
+	Path     string `yaml:"path,omitempty"`
+	Branch   string `yaml:"branch,omitempty"`
+}
+
+type Environment struct {
+	Region          string  `yaml:"region"`
+	Organization    string  `yaml:"organization"`
+	StackName       string  `yaml:"stackName"`
+	BaseProject     Project `yaml:"baseProject"`
+	PlatformProject Project `yaml:"platformProject"`
+	AppProject      Project `yaml:"appProject"`
+	DataProject     Project `yaml:"dataProject"`
+}
+
+func main() {
+	// Determine mode of operation; default is to refresh/update
+	destroy := false
+	argsWithoutProg := os.Args[1:]
+	if len(argsWithoutProg) > 0 {
+		if argsWithoutProg[0] == "destroy" {
+			destroy = true
+		}
+	}
```

Comment:
This is a style nit. You make take it or leave it.
```suggestion
	var destroy bool
	argsWithoutProg := os.Args[1:]
	if len(argsWithoutProg) > 0 {
		destroy = argsWithoutProg[0] == "destroy"
	}
```